[python] Python Integration Mock API Tests fail due to azure-http-specs version skew (pinned alpha.43 mock vs client regenerated from typespec-azure main)

## Summary

The **Python Integration → Mock API Tests** check (workflow `.github/workflows/python-integration.yml`) can fail with a body-validation error that is **unrelated to the PR under test**, due to a version skew between:

- the **azure-http-specs mock server** (pinned to `@azure-tools/azure-http-specs@0.1.0-alpha.43` via `http-client-python`), and
- the **generated client**, which is regenerated from the **`Azure/typespec-azure` `main` `.tsp` source** checked out by the workflow.

When the spec source on `typespec-azure` `main` is newer than the pinned `alpha.43` bundle, the client and the mock disagree and the tests fail.

## Observed failure

Seen on PR #11272 (a change that only touches content-type header serialization, unrelated to ARM common-properties):

Run: https://github.com/microsoft/typespec/actions/runs/29883405836/job/88810730906

```
FAILED mock_api/azure/test_azure_arm_commonproperties.py::test_arm_resource_identifiers_create_or_replace
FAILED mock_api/azure/test_azure_arm_commonproperties.py::test_arm_resource_identifiers_get
FAILED mock_api/azure/asynctests/test_azure_arm_commonproperties_async.py::test_arm_resource_identifiers_create_or_replace
FAILED mock_api/azure/asynctests/test_azure_arm_commonproperties_async.py::test_arm_resource_identifiers_get

azure.core.exceptions.HttpResponseError: (None) Body provided doesn't match expected body:
  at $.properties: Key count mismatch: expected 4 but got 5. extra: [armIdWithGroupScope]
```

The generated client sends **5** `properties` (including `armIdWithGroupScope`); the mock server expects **4**.

## Root cause

The `Build & Regenerate` job pins only the **compiled mock `/dist`** to the http-client-python-resolved `alpha.43`, but the **client is regenerated from the freshly checked-out `typespec-azure` `main` `.tsp` source**:

- Mock server dist is overwritten with the pinned `alpha.43` bundle:
  https://github.com/microsoft/typespec/blob/f7ab08305ae639ead6972c0e65007d6ef5ff9b33/.github/workflows/python-integration.yml#L70-L75
- Client is regenerated right after, from the checked-out `typespec-azure` source:
  https://github.com/microsoft/typespec/blob/f7ab08305ae639ead6972c0e65007d6ef5ff9b33/.github/workflows/python-integration.yml#L81-L83

The pinned version:
- https://github.com/microsoft/typespec/blob/f7ab08305ae639ead6972c0e65007d6ef5ff9b33/packages/http-client-python/package.json#L111
  ```json
  "@azure-tools/azure-http-specs": "0.1.0-alpha.43",
  ```

Meanwhile, `armIdWithGroupScope` was added to **both** the model and the mock in `typespec-azure` `main` by PR Azure/typespec-azure#4891 (2026-07-15), which is **newer than the `alpha.43` bundle**:

- `.tsp` model (5th property):
  https://github.com/Azure/typespec-azure/blob/1081f0678431238b3451fdfd82dce3679e572f21/packages/azure-http-specs/specs/azure/resource-manager/common-properties/arm-resource-identifier.tsp#L68
- mock expected body (create-or-replace):
  https://github.com/Azure/typespec-azure/blob/1081f0678431238b3451fdfd82dce3679e572f21/packages/azure-http-specs/specs/azure/resource-manager/common-properties/mockapi.ts#L277-L296

So within `typespec-azure` `main` the spec is internally consistent (model **and** mock both have 5 fields). The skew is introduced **only by the workflow**, which mixes:
- **client** = `typespec-azure` `main` source (post-#4891 → 5 fields), and
- **mock** = pinned `alpha.43` dist (pre-#4891 → 4 fields).

## Impact

Any PR that only touches `packages/http-client-python/**` (which is what triggers this workflow) can get a **red Mock API Tests check** it did not cause, whenever `typespec-azure` `main` advances an `azure-http-specs`/`http-specs` spec beyond the pinned `alpha.43` bundle. This blocks otherwise-green PRs (e.g. #11272).

## Suggested fixes (pick one)

1. **Bump the pin** in `packages/http-client-python/package.json` for `@azure-tools/azure-http-specs` (and `@typespec/http-specs`) to a version that includes Azure/typespec-azure#4891, so the mock dist matches the spec source used for regeneration.
2. **Regenerate the client from the same pinned bundle** used for the mock, instead of from the checked-out `typespec-azure` `main` source — i.e. make both sides use one source of truth in `python-integration.yml`.
3. **Do not pin the mock dist to the http-client-python-resolved version** — build/use the mock from the same checked-out `typespec-azure` source used for regeneration, keeping client and mock in lockstep.

Option 2 or 3 removes the class of failure entirely; option 1 fixes the current occurrence.

## Notes

- The `http-client-python` package's own unit/e2e tests (`typespec - ci` on Azure DevOps) are unaffected and pass; this is specific to the downstream `typespec-python` (`Azure/typespec-azure`) Mock API e2e run wired up by `python-integration.yml`.

--- a/.github/workflows/python-integration.yml
+++ b/.github/workflows/python-integration.yml
@@ -67,12 +67,15 @@ jobs:
       - name: Build
         run: pnpm turbo run --filter "@azure-tools/typespec-python..." build
 
-      - name: Use pinned http-specs versions
-        run: |
-          rm -rf core/packages/http-specs/dist
-          cp -r core/packages/http-client-python/node_modules/@typespec/http-specs/dist core/packages/http-specs/dist
-          rm -rf packages/azure-http-specs/dist
-          cp -r core/packages/http-client-python/node_modules/@azure-tools/azure-http-specs/dist packages/azure-http-specs/dist
+      - name: Build http-specs and azure-http-specs from source
+        # Build the mock servers from the checked-out typespec-azure (+ PR core)
+        # source, so the mock stays in lockstep with the client that is
+        # regenerated from the same source below. Previously these build outputs
+        # were overwritten with the versions resolved via http-client-python's
+        # package.json pin, which caused client/mock spec skew whenever
+        # typespec-azure main advanced a spec beyond the pinned bundle and failed
+        # otherwise-green PRs. See microsoft/typespec#11348.
+        run: pnpm turbo run build --filter "@typespec/http-specs..." --filter "@azure-tools/azure-http-specs..."
 
       - name: Prepare Python environment
         run: pnpm run prepare

